src/test/java/net/snowflake/ingest/streaming/internal/datatypes/AbstractDataTypeTest.java

```diff
@@ -575,17 +595,56 @@ protected void testIcebergIngestAndQuery(
     for (Object expectedValue : expectedValues) {
       Assertions.assertThat(resultSet.next()).isTrue();
       Object res = resultSet.getObject(1);
-      if (expectedValue instanceof BigDecimal) {
-        Assertions.assertThat(res)
-            .usingComparatorForType(BigDecimal::compareTo, BigDecimal.class)
-            .usingRecursiveComparison()
-            .isEqualTo(expectedValue);
-      } else if (expectedValue instanceof Timestamp) {
-        Assertions.assertThat(res.toString()).isEqualTo(expectedValue.toString());
-      } else {
-        Assertions.assertThat(res).isEqualTo(expectedValue);
-      }
+      assertEqualValues(expectedValue, res);
     }
     Assertions.assertThat(resultSet.next()).isFalse();
   }
+
+  protected void verifyMultipleColumns(
+      String tableName,
+      List<Map<String, Object>> values,
+      List<Map<String, Object>> expectedValues,
+      String orderBy)
+      throws Exception {
+    SnowflakeStreamingIngestChannel channel = openChannel(tableName);
+    Set<String> keySet = new HashSet<>();
+    for (Map<String, Object> value : values) {
+      String offsetToken = UUID.randomUUID().toString();
+      channel.insertRow(value, offsetToken);
+      TestUtils.waitForOffset(channel, offsetToken);
+    }
+
+    for (Map<String, Object> value : expectedValues) {
+      keySet.addAll(value.keySet());
+    }
+
+    for (String key : keySet) {
+      String query = String.format("select %s from %s order by %s", key, tableName, orderBy);
+      ResultSet resultSet = conn.createStatement().executeQuery(query);
```

Comment:
If there are 20 columns, we are doing select 20 times in here. Looks like it should be possible to do a single select and still do all the right checks against expectedValues?

Reply:
Yes. Changed to single select.